### ISSUE TITLE
refresh contour + cert-manager guide

### ISSUE THREAD

**Issue by hargrave81** [projectcontour/contour#4095]
I need help implementing the cert manager - I have successfully done this before, but I had to blow away my kubernetes cluster and rebuild it, and version 1.22 does not support the v1beta1 Ingress networking api version among other things.  So let me explain a little.

https://projectcontour.io/guides/cert-manager/

Guide above if you follow it, it doesn't work, it breaks when you attempt to use the jetstacks cert-manager part.  It never successfully gets a "test" cert.  Following the guide it gives you an error that is TLS handshake related.  I think that the v1.1.0 is just outdated and not making the cut, so I installed it via helm charts their most updated flavor.  BINGO, cert manager installed and took the cert file. 

[_basically trying to apply -f letsencrypt-staging.yaml kept failing with the kubectl command saying the TLS handshake was not good, I tried following several things online related to telling the OS to trust a different root since lest encrypt changed roots this past year, but that wasn't apparently the real issue_]

So once installing it via HELM, things started progressing, I was able to get the staging yaml in, and onward we went, but then the issuance of certs broke down.

so far - Follow the guide to install contour - 
1.5.4 cert manager
1.22 kubernetes cluster

No certs being issued

Contour logs catch this error: 

level=error msg="unresolved secret reference" context=IngressProcessor error="Secret not found" name=homepagepmry namespace=default secret=default/homepagepmry

Cert manager fails because the proxy isn't listening: 

cert-manager/controller/challenges "msg"="propagation check failed" "error"="failed to perform self check GET request 'http://imgdelivered.com/.well-known/acme-challenge/4yOakFzq6XGgNAISChQuezxya8emsF20NtMLO6b03Nk': Get \"http://imgdelivered.com/.well-known/acme-challenge/4yOakFzq6XGgNAISChQuezxya8emsF20NtMLO6b03Nk\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)" "dnsName"="imgdelivered.com" "resource_kind"="Challenge" "resource_name"="homepagepmry-fckbt-2465372399-3808274175" "resource_namespace"="default" "resource_version"="v1" "type"="HTTP-01"
I1011 22:22:53.090974 1 controller.go:161] cert-manager/controller/challenges "msg"="re-queuing item due to optimistic locking on resource" "key"="default/homepagepmry-fckbt-2465372399-3808274175" "error"="Operation cannot be fulfilled on challenges.acme.cert-manager.io \"homepagepmry-fckbt-2465372399-3808274175\": the object has been modified; please apply your changes to the latest version and try again"
I1011 22:22:53.091119 1 pod.go:59] cert-manager/controller/challenges/http01/selfCheck/http01/ensurePod "msg"="found one existing HTTP01 solver pod" "dnsName"="imgdelivered.com" "related_resource_kind"="Pod" "related_resource_name"="cm-acme-http-solver-hmmm5" "related_resource_namespace"="default" "related_resource_version"="v1" "resource_kind"="Challenge" "resource_name"="homepagepmry-fckbt-2465372399-3808274175" "resource_namespace"="default" "resource_version"="v1" "type"="HTTP-01"

The secrets are created for example: "homepagepmry-cwdcm" as Opaque with 1 value in it as of this moment of previewing it - tls.key in the default name space where the service resides

Hopefully you can help me connect the dots.

**Comment by stevesloka** [comment]
Hey @hargrave81, sorry for the troubles here! I actually ran into something similar on my cluster as well. Current Contour doesn't support Ingress/v1beta1 any longer, so if you have an older cert-manager, the HTTP01 challenge never succeeds and you don't get the secret in Kubernetes. 

Upgrading to the latest cert-manager (I think I used v1.5) fixed up my issues. We need to go back and update that guide to make sure its up to date.
